Cannot open remote development modal after project change

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

1. Open any project in Zed
2. Open any other project in Zed
3. Open the Recent Project picker and click "Connect..."
4. Notice that the dev server config modal is no longer accessible

### Environment

Zed: v0.135.0 (Zed Nightly 89039f6f34910fadce76c73258b731f0a016ccd6)
OS: macOS 14.4.1
Memory: 18 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature


https://github.com/zed-industries/zed/assets/16290753/d6bad0be-a28e-4ae7-91af-69b2c15f6ef3



### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

```bash
[INFO] starting language servers for Rust: rust-analyzer
[INFO] starting language servers for Rust: rust-analyzer
[ERROR] crates/project/src/project.rs:3398: no worktree found for diagnostics path "/Users/<username>/.cargo/registry/src/index.crates.io-6f17d22bba15001f/itertools-0.11.0/src/lib.rs"
```

--- a/crates/recent_projects/src/dev_servers.rs
+++ b/crates/recent_projects/src/dev_servers.rs
@@ -2,6 +2,7 @@ use std::time::Duration;
 
 use dev_server_projects::{DevServer, DevServerId, DevServerProject, DevServerProjectId};
 use editor::Editor;
+use feature_flags::FeatureFlagAppExt;
 use feature_flags::FeatureFlagViewExt;
 use gpui::{
     percentage, Action, Animation, AnimationExt, AnyElement, AppContext, ClipboardItem,
@@ -49,15 +50,23 @@ enum Mode {
 }
 
 impl DevServerProjects {
-    pub fn register(_: &mut Workspace, cx: &mut ViewContext<Workspace>) {
+    pub fn register(workspace: &mut Workspace, cx: &mut ViewContext<Workspace>) {
         cx.observe_flag::<feature_flags::Remoting, _>(|enabled, workspace, _| {
             if enabled {
-                workspace.register_action(|workspace, _: &OpenRemote, cx| {
-                    workspace.toggle_modal(cx, |cx| Self::new(cx))
-                });
+                Self::register_open_remote_action(workspace);
             }
         })
         .detach();
+
+        if cx.has_flag::<feature_flags::Remoting>() {
+            Self::register_open_remote_action(workspace);
+        }
+    }
+
+    fn register_open_remote_action(workspace: &mut Workspace) {
+        workspace.register_action(|workspace, _: &OpenRemote, cx| {
+            workspace.toggle_modal(cx, |cx| Self::new(cx))
+        });
     }
 
     pub fn open(workspace: View<Workspace>, cx: &mut WindowContext) {

